tests/lib/kubernetes/caasp.py

```diff
@@ -34,29 +34,29 @@ def __init__(self, hardware):
         self._clusterpath = os.path.join(hardware.working_dir, 'cluster')
         self._kubeconfig = os.path.join(self.hardware.working_dir,
                                         'admin.conf')
+        self._ssh_agent_auth_sock = os.path.join(hardware.working_dir,
+                                                 'ssh-agent.sock')
         self._ssh_agent()
 
     def _ssh_agent(self):
         try:
-            res = subprocess.run(['ssh-agent'], check=True,
-                                 capture_output=True)
+            res = subprocess.run(
+                ['ssh-agent', '-a', self._ssh_agent_auth_sock],
+                check=True, capture_output=True)
         except subprocess.CalledProcessError:
-            msg = 'Failed to start ssh agent'
-            logger.exception(msg)
+            logger.exception('Failed to start ssh agent')
             raise
 
-        self.SSH_AUTH_SOCK = res.stdout.decode(
-            'utf-8').split(';')[0].split('=')[1]
-        self.SSH_AGENT_PID = res.stdout.decode(
+        logger.info(f"ssh-agent started: {res.stdout.decode('utf-8')}")
+        self._ssh_agent_pid = res.stdout.decode(
             'utf-8').split(';')[2].split('=')[1]
-        os.environ['SSH_AUTH_SOCK'] = self.SSH_AUTH_SOCK
-        os.environ['SSH_AGENT_PID'] = self.SSH_AGENT_PID
+        os.environ['SSH_AUTH_SOCK'] = self._ssh_agent_auth_sock
+        os.environ['SSH_AGENT_PID'] = self._ssh_agent_pid
```

Comment:
Since we're refactoring the ssh-agent, I would still prefer that we don't modify env-vars and instead pass them into subprocess.

Reply:
yes. agreed. but I would do that in another commit/PR if that's fine for you

Reply:
Sure :-)